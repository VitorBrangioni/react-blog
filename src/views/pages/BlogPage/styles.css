html,
body {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  background: white;
}

html {
  /* 1rem = 10px */
  font-size: 62.5%;
}

main {
  padding: 60px;
  background: white;
  grid-column-start: 1;
  grid-column-end: 2;
  width: 100%;
  grid-row-start: 2;
  grid-row-end: null;
}

h2 {
  color: white;
  font-size: 2.6rem;
}

h3 {
  color: white;
  font-size: 1.6rem;
}

.container {
  display: grid;
  grid-template-columns: 3fr 1fr;
  grid-template-rows: 20vh 80vh;
}

*,
*::before,
*::after {
  box-sizing: border-box;
}

h1 {
  font-size: 3.6rem;
  font-weight: 100;
  letter-spacing: 3px;
  padding-top: 5px;
  color: "white";
  padding-bottom: 5px;
  text-transform: uppercase;
}

p {
  font-size: 1.6rem;
}
