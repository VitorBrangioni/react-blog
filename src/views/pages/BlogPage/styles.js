import styled from "styled-components";

export const Main = styled.main`
  padding: 60px;
  background: white;
  grid-column-start: 1;
  grid-column-end: 2;
  width: 100%;
  grid-row-start: 2;
  grid-row-end: null;
`;

export const Container = styled.div`
  display: grid;
  grid-template-columns: 3fr 1fr;
  grid-template-rows: 20vh 80vh;
`;

export const Aside = styled.aside`
  background: #871ae5;
  grid-column: 2/3;
  grid-row: 2 / null;
  padding: 60px 30px;
`;
