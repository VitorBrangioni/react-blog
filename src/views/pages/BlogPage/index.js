import React, { Component } from "react";
import "./styles.css";
import Header from "../../components/Header";
import PostsList from "../../components/PostsList";
import RecentPosts from "../../components/RecentPosts";
import { Main, Container, Aside } from "./styles";

export class BlogPage extends Component {
  render() {
    return (
      <Container>
        <Header />
        <Main>
          <PostsList />
        </Main>
        <Aside>
          <RecentPosts />
        </Aside>
      </Container>
    );
  }
}

export default BlogPage;
