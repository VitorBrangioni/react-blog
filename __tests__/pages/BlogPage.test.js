import React from "react";
import BlogPage from "../../src/views/pages/BlogPage";
import { shallow } from "enzyme";

describe("BlogPage", () => {
  describe("BlogPage", () => {
    it("Snapshot test", () => {
      const wrapper = shallow(<BlogPage />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
