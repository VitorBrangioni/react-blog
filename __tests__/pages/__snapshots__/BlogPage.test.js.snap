// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPage BlogPage Snapshot test 1`] = `
<styled.div>
  <HeaderComponent />
  <styled.main>
    <Connect(PostsList) />
  </styled.main>
  <styled.aside>
    <Connect(PostsListFilter) />
    <Connect(RecentPosts) />
  </styled.aside>
</styled.div>
`;

exports[`BlogPage Snapshot test 1`] = `
<styled.div>
  <HeaderComponent />
  <styled.main>
    <Connect(PostsList) />
  </styled.main>
  <styled.aside>
    <Connect(PostsListFilter) />
    <Connect(RecentPosts) />
  </styled.aside>
</styled.div>
`;
